### PR TITLE
Add mruby and quickjs + allow to specify benchmarks in exp requests

### PR DESCRIPTION
**Pull request by andreafioraldi**
Hi!
This is a contribution PR.
In order to allow the comparison of grammar fuzzers on fuzzbench, I added two engines that parse popular programming languages often used to evaluate those fuzzers, ruby and javascript.
I choose quickjs and mruby as they are quick to compile (the CI will be happy) and already on oss-fuzz.
In addition, I added a feature to the experiment requests, allowing the user to specify a selection of benchmarks in the yaml file (under `benchmarks:` as we already do for fuzzers), to allow experiments with grammar fuzzers to pick just these targets.
PHP is another popular benchmark, and it's already in, so with this PR fuzzbench will have 3 subjects for grammar fuzzing evaluation. LUA and some C-like compilers like tcc and solidity are other options that can be considered in the future.

I will soon (after the end of the experiment #1327 for sure) submit a first experiment comparing 4 different grammar-based techniques (some of them never evaluated against each other) vs. a generic bit-level fuzzer to test the benchmarks.

I'm sure that mruby is vulnerable, it is a pre-ossfuzz commit, but for quickjs I just picked one of the first releases. It is vulnerable too with an high probability.

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
> I will soon (after the end of the experiment #1327 for sure) submit a first experiment comparing 4 different grammar-based techniques (some of them never evaluated against each other) vs. a generic bit-level fuzzer to test the benchmarks.
> 
> I'm sure that mruby is vulnerable, it is a pre-ossfuzz commit, but for quickjs I just picked one of the first releases. It is vulnerable too with an high probability.

We've got php parsers in fuzzbench, can try this too.

**Review by jonathanmetzman — COMMENTED** [review]
LGTM with nits

File: benchmarks/mruby-2018-05-23/mruby_fuzzer.options [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[libfuzzer]
```
[the remark this answers is not in the document]
Comment (andreafioraldi) [reply benchmarks/mruby-2018-05-23/mruby_fuzzer.options:1]: It is used in other benchmarks too

File: service/automatic_run_experiment.py [hunk 2]
```diff
@@ -212,9 +212,23 @@ def run_requested_experiment(dry_run):
...
+
+    benchmarks = requested_experiment.get('benchmarks')
+    if benchmarks is None:
+        benchmarks = valid_benchmarks
+    else:
+        errors = False
+        for bench in benchmarks:
+            if bench not in valid_benchmarks:
+                logs.error(
+                    'Requested experiment: in %s, %s is not a valid %s ' \
```
[the remark this answers is not in the document]
Comment (andreafioraldi) [reply service/automatic_run_experiment.py:227]: iirc presubmit was failing with line too long

File: benchmarks/mruby-2018-05-23/mruby_fuzzer.options (hunk 1 above)
[the remark this answers is not in the document]
Comment (andreafioraldi) [reply benchmarks/mruby-2018-05-23/mruby_fuzzer.options:1]: should I remove from all of them? fork = 1 maybe is needed

File: service/automatic_run_experiment.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply service/automatic_run_experiment.py:227]: I think this should work:
```
logs.error('Requested experiment: in %s, %s is not a valid %s '
                'benchmark.', requested_experiment, bench, benchmark_type)
```

[the remark this answers is not in the document]
Comment (andreafioraldi) [reply service/automatic_run_experiment.py:227]: yess `service/automatic_run_experiment.py:227:0: C0301: Line too long (83/80) (line-too-long)`

[the remark this answers is not in the document]
Comment (andreafioraldi) [reply service/automatic_run_experiment.py:227]: make format is not enough to make presubmit happy

[the remark this answers is not in the document]
Comment (andreafioraldi) [reply service/automatic_run_experiment.py:227]:
```python
        for benchmark in benchmarks:
            if benchmark not in valid_benchmarks:
                logs.error(
                    'Requested experiment: in %s, %s is not a valid %s benchmark.',
                    requested_experiment, benchmark, benchmark_type)
                errors = True
        if errors:
            return
```

this is not ok for presubmit

File: benchmarks/mruby-2018-05-23/mruby_fuzzer.options (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply benchmarks/mruby-2018-05-23/mruby_fuzzer.options:1]: > It is used in other benchmarks too

You mean other benchmarks have this file?
I think they have it when the dictionary isn't named $FUZZ_TARGET.dict

[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply benchmarks/mruby-2018-05-23/mruby_fuzzer.options:1]: I don't think fork=1 actually will be used anywhere.

File: service/automatic_run_experiment.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (andreafioraldi) [reply service/automatic_run_experiment.py:227]:
```python
                logs.error(
                    'Requested experiment:' \
                    ' in %s, %s is not a valid %s benchmark.',
                    requested_experiment, benchmark, benchmark_type)
```
this is ok

[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply service/automatic_run_experiment.py]: Need to do this reinstall anyways

[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply service/automatic_run_experiment.py]: So this is valid according to my local use of `make format`:
```
                logs.error(
                    'Requested experiment: in %s, %s is not a valid %s '
                    'benchmark.', requested_experiment, bench, benchmark_type)
```

[the remark this answers is not in the document]
Comment (andreafioraldi) [reply service/automatic_run_experiment.py]: got it, I was not aware of multiline strings in python without backlash. I'm too used to C and Rust

**Review by jonathanmetzman — APPROVED** [review]
LGTM

**Comment by andreafioraldi** [comment]
> OK, is it good to merge now then?

Seems yes, I cannot really say if it works for the failing fuzzers because they fail for other reasons but it should

**Pull request merged**